reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogHandler.java

```diff
@@ -14,39 +14,45 @@
  * limitations under the License.
  */
 
-package reactor.netty.http.server;
+package reactor.netty.http.server.logging;
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufHolder;
-import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelPromise;
-import io.netty.channel.socket.SocketChannel;
 import io.netty.handler.codec.http.HttpRequest;
 import io.netty.handler.codec.http.HttpResponse;
 import io.netty.handler.codec.http.HttpResponseStatus;
 import io.netty.handler.codec.http.HttpUtil;
 import io.netty.handler.codec.http.LastHttpContent;
+import reactor.util.annotation.Nullable;
+
+import java.util.function.Function;
 
 /**
+ * {@link ChannelHandler} for access log of HTTP/1.1.
+ *
  * @author Violeta Georgieva
+ * @author limaoning
  */
-final class AccessLogHandler extends ChannelDuplexHandler {
+public final class AccessLogHandler extends BaseAccessLogHandler {
```

Comment:
to maintain consistency, I'd rename this class to `AccessLogHandlerH1`.

@violetagg @limaoning I'm also wondering about the `public` status of this and the H2 implementations. maybe moving _everything_ in `logging` package wasn't such a good idea 😖 

I wonder if it wouldn't be better to put the concrete classes back into the base `http.server` package, as package-private, and only keep the interfaces and user-facing classes here in `http.server.logging`. ie classes that make sense for a user to implement/customize/inherit from

Reply:
> to maintain consistency, I'd rename this class to AccessLogHandlerH1.

Agree with that

And let's move to the `http.server` package ONLY the handlers `BaseAccessLogHandler`, `AccessLogHandlerH1`, `AccessLogHandlerH2`

Reply:
But `AccessLogArgProviderH1`, `AccessLogArgProviderH2` will be `public`.

Reply:
if we move them to `http.server` they can be package private - that's the idea

Reply:
You are saying move handlers to `http.server`, but `AccessLogArgProviderH1`, `AccessLogArgProviderH2` are not, and they will be `public`. Otherwise, move `AccessLogArgProviderH1`, `AccessLogArgProviderH2` to `http.server` too?

Reply:
> And let's move to the http.server package ONLY the handlers BaseAccessLogHandler, AccessLogHandlerH1, AccessLogHandlerH2

yes

Reply:
@simonbasle Here I think we either move everything on keep the handlers public  as @limaoning noted in the comment above

Reply:
Let me think a bit here

Reply:
some rough idea

```
public class AccessLogHandlerFactory {

	ChannelHandler newAccessLogHandler(@Nullable Function<AccessLogArgProvider, AccessLog> accessLog, boolean isHttp2) {
		return isHttp2 ? new AccessLogHandlerH2(accessLog) : new AccessLogHandlerH1(accessLog);
	}
}
```

wdyt?

Reply:
or may be even something like this https://github.com/reactor/reactor-netty/blob/master/reactor-netty-core/src/main/java/reactor/netty/transport/logging/AdvancedByteBufFormat.java

Reply:
Should be OK.